### DISCUSSION TITLE
i think we should get rid of the little popup on the graph when labels are shown

### DISCUSSION THREAD

**Discussion by jmazzitelli** [kiali/kiali#4699] [General]
When you are zoomed in the graph and you see the node labels, I don't like that little popup in the graph that delays popping up when you hover over something and delays going away when you hover off. The delay is distracting, and the little popup gives you no additional information you can't already find on the screen (and sometimes that popup covers up things in the UI).

But I like that popup when the labels are hidden (when you are zoomed out). Because it gives you information that you can't otherwise see on the screen.

What do people think about hiding that popup window when labels are shown on the graph?

This is what I am talking about:

![Peek 2022-02-01 17-08](https://user-images.githubusercontent.com/2029470/152061723-8145559b-5317-473c-989a-9f06f7ed0678.gif)

**Comment by jshaughn** [comment]
FYI, the "little popup" is called the "context info".  It's the same as the top of the "context menu".

**Comment by lucasponce** [comment]
I think this discussion refers to the current master branch.
I see some CSS differences when hovering comparing what I remember when this was consolidated, I'm not sure if perhaps is a style regression with the latest work.

> What do people think about hiding that popup window when labels are shown on the graph?

I think the context in the edges is useful, I don't see adding too much noise when labels are on the elements but I can understand that can be hidden in these cases.

But my proposal would add in the context info more valuable information rather than hide it, there are more flags (security, values) that can be also added in the context info and make it always present.

**Comment by nrfox** [comment]
The info does seem redundant when the labels are visible. At least it doesn't add anything to what you already know from either the labels or the shapes. It might be weird to have the hover action do different things depending on the zoom level but the context info is primarily there to help in scenarios where there are no labels right?

**Comment by jshaughn** [comment]
The more I play with this the more I think the contextInfo should be reserved to when labels are not available.  In my [PoC for edge work](https://github.com/kiali/kiali-ui/pull/2289) I've currently got this change and I am preferring the behavior.
